ibis/backends/mssql/tests/conftest.py

```diff
@@ -0,0 +1,121 @@
+MSSQL_PASS = os.environ.get('IBIS_TEST_MSSQL_PASSWORD', '1bis_Testing!')
+MSSQL_HOST = os.environ.get('IBIS_TEST_MSSQL_HOST', 'localhost')
+MSSQL_PORT = int(os.environ.get('IBIS_TEST_MSSQL_PORT', 1433))
+IBIS_TEST_MSSQL_DB = os.environ.get('IBIS_TEST_MSSQL_DATABASE', 'ibis_testing')
+
+
+class TestConf(BackendTest, RoundHalfToEven):
+    # MSSQL has the same rounding behavior as postgres
+    check_dtype = False
+    supports_window_operations = False
+    returned_timestamp_unit = 's'
+    supports_arrays = False
+    supports_arrays_outside_of_select = supports_arrays
+    bool_is_int = True
+    supports_structs = False
+    supports_arrays = False
+
+    def __init__(self, data_directory: Path) -> None:
+        super().__init__(data_directory)
+        # mariadb supports window operations after version 10.2
+        # but the sqlalchemy version string looks like:
+        # 5.5.5.10.2.12.MariaDB.10.2.12+maria~jessie
+        # or 10.4.12.MariaDB.1:10.4.12+maria~bionic
+        # example of possible results:
+        # https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_3/
+        # test/dialect/MSSQL/test_dialect.py#L244-L268
+        con = self.connection
+        if 'MariaDB' in str(con.version):
+            # we might move this parsing step to the MSSQL client
+            version_detail = con.con.dialect._parse_server_version(str(con.version))
+            version = (
+                version_detail[:3]
+                if version_detail[3] == 'MariaDB'
+                else version_detail[3:6]
+            )
+            self.__class__.supports_window_operations = version >= (10, 2)
+        elif parse_version(con.version) >= parse_version('8.0'):
+            # MSSQL supports window operations after version 8
+            self.__class__.supports_window_operations = True
+
+    @staticmethod
+    def _load_data(
+        data_dir: Path,
+        script_dir: Path,
+        user: str = MSSQL_USER,
+        password: str = MSSQL_PASS,
+        host: str = MSSQL_HOST,
+        port: int = MSSQL_PORT,
+        database: str = IBIS_TEST_MSSQL_DB,
+        **_: Any,
+    ) -> None:
+        """Load test data into a MSSQL backend instance.
+
+        Parameters
+        ----------
+        data_dir
+            Location of testdata
+        script_dir
+            Location of scripts defining schemas
+        """
+        with open(script_dir / 'schema' / 'mssql.sql') as schema:
+            engine = init_database(
+                url=sa.engine.make_url(
+                    f"mssql+pymssql://{user}:{password}@{host}:{port:d}"
+                ),
+                database=database,
+                schema=schema,
+                isolation_level="AUTOCOMMIT",
+            )
+            with engine.begin() as con:
+                for table in TEST_TABLES:
+                    csv_path = data_dir / f"{table}.csv"
+                    lines = [
+                        f"LOAD DATA LOCAL INFILE {str(csv_path)!r}",
```

Comment:
Haven't gotten this working just yet.

Reply:
I keep getting "Adaptive Server is unavailable or does not exist", but I can use the same url to make an engine in sqlalchemy outside of the nix-shell environment...

Reply:
I was able to get this working by setting up a volume mount to `/data` from `ci/ibis-testing-data` and using `BULK INSERT`.

Reply:
The volume mount is set up in `docker-compose.yml`, so this should be usable in CI and developers